frontend/class-frontend-page-type.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * @package WPSEO\Frontend
+ */
+
+/**
+ * Represents the classifier for determine the current opened page type.
+ */
+class WPSEO_Frontend_Page_Type {
+
+	/**
+	 * @var WPSEO_WooCommerce_Shop_Page
+	 */
+	protected $woocommerce_shop_page;
+
+	/**
+	 * Sets the WooCommmerce shop page object.
+	 *
+	 * @param WPSEO_WooCommerce_Shop_Page $woocommerce_shop_page The woocommerce shop page object.
+	 */
+	public function __construct( WPSEO_WooCommerce_Shop_Page $woocommerce_shop_page ) {
+		$this->woocommerce_shop_page = $woocommerce_shop_page;
+	}
+
+	/**
+	 * Checks if the current opened page is singular.
+	 *
+	 * @return bool True when singular.
+	 */
+	public function is_singular() {
```

Comment:
With naming this conflicts conceptually with the WordPress' `is_singular`. I can't think of a better name, maybe `is_singular_indexable`?

Or even: `is_simple_page`/ `is_simple_indexable`, signifying that pages of this kind can be handled by simply using the data available in meta.